brave/src/main/java/brave/propagation/ExtraFieldPropagation.java

```diff
@@ -199,8 +200,10 @@ public Factory build() {
       for (i = 0; i < keyToField.length; i++) {
         keyToField[i] = keyToFieldList.get(i);
       }
-      return new Factory(delegate, fields.toArray(new String[0]), keys.toArray(new String[0]),
-        keyToField, redacted);
+      String[] validated = fields.toArray(new String[0]);
```

Comment:
Since we're not doing any extra validation here this is confusing to me. How about

```java
if (fields.isEmpty()) return new Factory(delegate);
return new RealFactory(delegate, fields.toArray(...
```

Reply:
sgtm